restore/restore.go

```diff
@@ -140,27 +142,155 @@ func (c *RestoreCommand) StartRestore(cliConnection plugin.CliConnection, servic
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 
-	if resp.Status != "202 Accepted" {
-		fmt.Println(AddColor("FAILED", red))
-		var message string = string(body)
-		var parts []string = strings.Split(message, ":")
-		fmt.Println("Error - ",parts[3])
+	var respObject map[string]interface{}
+
+	if err := json.Unmarshal(body, &respObject); err != nil {
+		fmt.Println(err)
+	}
+
+	if respStatus, flag := respObject["status"].(float64); flag != false {
+		if respStatus != 202 {
+			fmt.Println(AddColor("FAILED", red))
+			if respError, flag := respObject["error"].(string); flag != false {
+				fmt.Println("Error: ", respError)
+			}
+			if respMessage, flag := respObject["description"].(string); flag != false {
+				fmt.Println("Message: ", respMessage)
+			}
+		}
 	}
 
 	if resp.Status == "202 Accepted" {
 		fmt.Println(AddColor("OK", green))
-		if isGuidOperation  == true {
-		fmt.Println("Restore has been initiated for the instance name:", AddColor(serviceInstanceName, cyan), " and from the backup id:", AddColor(backupId, cyan))
+		if respOperation, flag := respObject["name"].(string); flag != false {
+			fmt.Println("Operation: ", respOperation)
+		}
+		if restoreGuid, flag := respObject["guid"].(string); flag != false {
+			fmt.Println("Restore Guid: ", restoreGuid)
+		}
+		if isGuidOperation == true {
+			fmt.Println("Restore has been initiated for the instance name:", AddColor(serviceInstanceName, cyan), " and from the backup id:", AddColor(backupId, cyan))
```

Comment:
IMHO ` and from the backup id:` is not required. `Restore has been initiated for the instance <ServiceInstanceName>`  is enough after triggering restore. Further technical demarkation is not required. From customer's perspective she is getting restore done. That's it.

Reply:
This is a legacy code.

Still, understanding the code here, I think what you are saying above would have been fine if we/customer wouldn't be passing any `backup-guid` in the start-restore call.  However, the customer is anyway needed to pass the `backup-guid` to the start-restore command. So, the response should make sense for him as well.

Reply:
Thought more from customer's perspective. I think only a simple success message is fine as customer already knows the details she entered for triggering restore, for simplicity and for better user experience. If there are 5 different ways to restore, then 5 different custom message in response. Moreover in backend server we don't do anything beyond customer's input, throw back error if doesn't match customer's input.

 Not a must fix. Leaving up to you 👍

Reply:
The initial idea was to keep it similar to cf commands.
e.g. for `cf create-service` command, the response from cf is `Creating service <INSTANCE_NAME> for service <SERVICE_NAME>/plan <PLAN_NAME> for org <ORG_NAME> and space <SPACE_NAME>`.

However, if you think this is the better approach, I will remove the backup-guid from the output response.

Reply:
ok go ahead with existing one (y)